### PR TITLE
fixes in capsule upgrade playbook tests

### PR DESCRIPTION
**Pull request by pondrejk**
fixing some test failures in tests related to capsule upgrade playbook -- the capsule target version is no longer hardcoded, incorect hostname and credentials selection is also fixed

### CODE REVIEW

**Review by Griffin-Sullivan — APPROVED** [review]

File: tests/foreman/api/test_remoteexecution.py [hunk 1]
```diff
@@ -19,9 +19,9 @@
 import pytest
 
 from robottelo.api.utils import wait_for_tasks
+from robottelo.hosts import get_sat_version
 
-
-CAPSULE_TARGET_VERSION = '6.10.z'
+CAPSULE_TARGET_VERSION = f'6.{get_sat_version().minor - 1}.z'
```
Comment (Gauravtalreja1) [inline tests/foreman/api/test_remoteexecution.py:24]: -1 here means target_version will be 6.11.z, shouldn't it be 6.12.z instead?

**Comment by jyejare** [comment]
@pondrejk Could you add some background of the issue in description to know the issue for reviewers.

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_remoteexecution.py -k capsule_upgrade_playbook

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_remoteexecution.py -k capsule_upgrade_playbook

**Comment by pondrejk** [comment]
trigger: test-robottelo
pytest: tests/foreman/destructive/test_remoteexecution.py -k capsule_upgrade_playbook

**Comment by pondrejk** [comment]
@jyejare @Gauravtalreja1 mind revisiting?

File: tests/foreman/destructive/test_remoteexecution.py [hunk 2]
```diff
@@ -58,16 +59,20 @@ def test_negative_run_capsule_upgrade_playbook_on_satellite(target_sat):
...
+            'status=failed,description="Capsule Upgrade Playbook"'
+        }
     )[0]
+    host = target_sat.api.Host().search(query={'search': target_sat.hostname})
     response = client.get(
-        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{sat.id}',
-        auth=(target_sat.username, target_sat.password),
+        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{host[0].id}',
+        auth=(settings.server.admin_username, settings.server.admin_password),
         verify=False,
```
Comment (Gauravtalreja1) [inline tests/foreman/destructive/test_remoteexecution.py:76]: why not implement this endpoint in nailgun and use it here with target_sat? (non-blocking)

File: tests/foreman/destructive/test_remoteexecution.py [hunk 3]
```diff
@@ -58,16 +59,20 @@ def test_negative_run_capsule_upgrade_playbook_on_satellite(target_sat):
...
+            'search': f'host={target_sat.hostname},'
+            'status=failed,description="Capsule Upgrade Playbook"'
+        }
     )[0]
+    host = target_sat.api.Host().search(query={'search': target_sat.hostname})
     response = client.get(
-        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{sat.id}',
-        auth=(target_sat.username, target_sat.password),
+        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{host[0].id}',
+        auth=(settings.server.admin_username, settings.server.admin_password),
```
Comment (Gauravtalreja1) [inline tests/foreman/destructive/test_remoteexecution.py:75]: You can use `get_credentials` func from config to fetch this

**Review by Gauravtalreja1 — COMMENTED** [review]
@pondrejk LGTM, Thanks for the updates, few suggestions

**Review by vsedmik — APPROVED** [review]

File: tests/foreman/destructive/test_remoteexecution.py (hunk 2 above)
Comment (pondrejk) [reply tests/foreman/destructive/test_remoteexecution.py:76]: good idea, I filed https://github.com/SatelliteQE/nailgun/issues/849 to track this

File: tests/foreman/destructive/test_remoteexecution.py [hunk 4]
```diff
@@ -58,16 +59,20 @@ def test_negative_run_capsule_upgrade_playbook_on_satellite(target_sat):
...
+        query={
+            'search': f'host={target_sat.hostname},'
+            'status=failed,description="Capsule Upgrade Playbook"'
+        }
     )[0]
+    host = target_sat.api.Host().search(query={'search': target_sat.hostname})
     response = client.get(
-        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{sat.id}',
-        auth=(target_sat.username, target_sat.password),
+        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{host[0].id}',
```
Comment (ogajduse) [inline tests/foreman/destructive/test_remoteexecution.py:74]: using the following patch, you can delete L72.
```suggestion
        f'{target_sat.url}/api/job_invocations/{job.id}/hosts/{target_sat.nailgun_host.id}',
```

**Review by ogajduse — APPROVED** [review]
One non-blocking suggestion.

File: tests/foreman/destructive/test_remoteexecution.py (hunk 4 above)
Comment (pondrejk) [reply tests/foreman/destructive/test_remoteexecution.py:74]: hm, for some reason this didn't work for me when I was trying that out locally

**Pull request merged**